dipy/align/_public.py

```diff
@@ -631,13 +448,36 @@ def affine_registration(moving, static,
                                 sigmas=sigmas,
                                 factors=factors)
 
+    if pipeline == ["center_of_mass"] and ret_metric:
+        raise ValueError("center of mass registration cannot return any "
+                         "quality metric.")
+
     # Go through the selected transformation:
     for func in pipeline:
-        starting_affine = func(moving, static,
-                               static_affine=static_affine,
-                               moving_affine=moving_affine,
-                               starting_affine=starting_affine,
-                               reg=affreg)
+        if func == "center_of_mass":
+            transform = transform_centers_of_mass(static, static_affine,
+                                                  moving, moving_affine)
+            starting_affine = transform.affine
+        else:
+            if func == "translation":
+                transform = TranslationTransform3D()
+            elif func == "rigid":
+                transform = RigidTransform3D()
+            elif func == "rigid_isoscaling":
+                transform = RigidIsoScalingTransform3D()
+            elif func == "rigid_scaling":
+                transform = RigidScalingTransform3D()
+            elif func == "affine":
+                transform = AffineTransform3D()
+            else:
+                raise ValueError("Not supported registration method")
```

Comment:
can you add a test for that? (`npt.assert_raises(ValueError, ...)`)

Reply:
done!